web/source/osk/visualKeyboard.ts

```diff
@@ -1660,8 +1660,7 @@ namespace com.keyman.osk {
     showKeyTip(key: KeyElement, on: boolean) {
       var tip = this.keytip;
 
-      // Do not change the key preview unless key or state has changed
-      if (tip == null || (key == tip.key && on == tip.state)) {
+      if (tip == null) {
```

Comment:
Was this change still needed after the change earlier in this file?  I find that interesting if so.

Reply:
From what I can recall, this was necessary because `showKeyTip` is called from a few places, and this was just removing the state check from an intermediate part of the state machine which was part of the desync issue. See my initial comment:

> But a more detailed look at the state machine and a refactoring would certainly be worthwhile.

😆